Content.Shared/DoAfter/DoAfterEventArgs.cs

```diff
@@ -0,0 +1,118 @@
+using System.Threading;
+using Content.Shared.FixedPoint;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.DoAfter;
+[Serializable, NetSerializable]
+public sealed class DoAfterEventArgs
+{
+    /// <summary>
+    ///     The entity invoking do_after
+    /// </summary>
+    public EntityUid User;
+
+    /// <summary>
+    ///     How long does the do_after require to complete
+    /// </summary>
+    public float Delay;
+
+    /// <summary>
+    ///     Applicable target (if relevant)
+    /// </summary>
+    public EntityUid? Target;
+
+    /// <summary>
+    ///     Entity used by the User on the Target.
+    /// </summary>
+    public EntityUid? Used;
+
+    public bool RaiseOnUser { get; set; } = true;
+
+    public bool RaiseOnTarget { get; set; } = true;
+
+    public bool RaiseOnUsed { get; set; } = true;
+
+    /// <summary>
+    ///     Manually cancel the do_after so it no longer runs
+    /// </summary>
+    [NonSerialized]
+    public CancellationToken CancelToken;
+
+    // Break the chains
+    /// <summary>
+    ///     Whether we need to keep our active hand as is (i.e. can't change hand or change item).
+    ///     This also covers requiring the hand to be free (if applicable).
+    /// </summary>
+    public bool NeedHand { get; set; }
+
+    /// <summary>
+    ///     If do_after stops when the user moves
+    /// </summary>
+    public bool BreakOnUserMove { get; set; }
+
+    /// <summary>
+    ///     If do_after stops when the target moves (if there is a target)
+    /// </summary>
+    public bool BreakOnTargetMove { get; set; }
+
+    /// <summary>
+    ///     Threshold for user and target movement
+    /// </summary>
+    public float MovementThreshold { get; set; }
+
+    public bool BreakOnDamage { get; set; }
```

Comment:
All of these properties should just be fields on DoAfter.cs

Reply:
Fixed in https://github.com/space-wizards/space-station-14/pull/13225/commits/01f3c93afeb160420735aa628babce9e90b52fd9